### PR TITLE
Disable nvrtc workaround

### PR DESCRIPTION
**Pull request by miscco**
nvbug3961621 has been fixed in 12.2

I was too lazy to create a new cudacc version for 12.2 so I just picked 12.3

Addresses nvbug4263883

### CODE REVIEW

**Review by griwes — APPROVED** [review]

**Review by wmaxey — APPROVED** [review]

**Review by gevtushenko — APPROVED** [review]

**Comment by leofang** [comment]
Q: what's the implication for people using older NVRTC? We have a wide range of support matrix in Python (CUDA 11/12) that could consume libcu++ headers at run time.

**Comment by miscco** [comment]
> Q: what's the implication for people using older NVRTC? We have a wide range of support matrix in Python (CUDA 11/12) that could consume libcu++ headers at run time.

NVRTC prior to 12.6 will still use the workaround

File: libcudacxx/include/cuda/std/detail/libcxx/include/__config [hunk 1]
```diff
@@ -865,21 +865,22 @@ typedef unsigned int char32_t;
...
 // This creates a whole lot of boilerplate that we can avoid through a macro (see nvbug3961621)
-#  if _CCCL_COMPILER(NVRTC) || (_CCCL_CUDACC_BELOW(11, 3) && _CCCL_COMPILER(CLANG))
-#    define _LIBCUDACXX_DELEGATE_CONSTRUCTORS(__class, __baseclass, ...)                               \
-      using __base = __baseclass<__VA_ARGS__>;                                                         \
-      template <class... _Args, enable_if_t<_CCCL_TRAIT(is_constructible, __base, _Args...), int> = 0> \
-      _LIBCUDACXX_HIDE_FROM_ABI constexpr __class(_Args&&... __args) noexcept(                         \
-        _CCCL_TRAIT(is_nothrow_constructible, __base, _Args...))                                       \
-          : __base(_CUDA_VSTD::forward<_Args>(__args)...)                                              \
-      {}                                                                                               \
+#  if (_CCCL_CUDACC_BELOW(12, 6) && _CCCL_COMPILER(NVRTC)) || (_CCCL_CUDACC_BELOW(11, 3) && _CCCL_COMPILER(CLANG))
```
Comment (miscco) [inline libcudacxx/include/cuda/std/detail/libcxx/include/__config:868]: @leofang Here you can see that the workaround is in place for NVRTC and CTK < 12.6 as well as clang and CTK below 11.3

Comment (leofang) [reply libcudacxx/include/cuda/std/detail/libcxx/include/__config:868]: Great, thanks for the pointer Michael!

**Pull request merged**